### PR TITLE
Remove material menu

### LINKED ISSUE #748: Removing material menu
After moving to react, many issues occured regarding the material (mdl) menu used in footer.
This is due to mdl not being connected with react lifecycle, forcing us to manually keep it up to date to dynamically update DOM and events.

One option is using `material-react`, but I would suggest to write our own since this is a very small and specific use case. Also would be easier to customise.

[Prerequisites]:
Using button instead of <a> to match react-a10y. 
Easy to customise CSS to match [Firefox Photon Doorhangers](https://design.firefox.com/photon/components/doorhangers.html)
Should fix #716,  #721

### PR DESCRIPTION
**Pull request by sebastienbarbier**
Replace material context menu with some native react code following Firefox Photon styling.

footer

Still need to implement :
- [x] Dark theme
- [x] Keyboard navigation

Fix #748

### CODE REVIEW

**Comment by sebastienbarbier** [comment]
@vladikoff Is keyboard navigation a requirement ?

**Comment by vladikoff** [comment]
> @vladikoff Is keyboard navigation a requirement ?

**Comment by vladikoff** [comment]
oops,

> @vladikoff Is keyboard navigation a requirement ?

Seems so, Softvision tests for it. Is it hard to add?  
On this demo page keyboard nav actually works: https://getmdl.io/components/index.html#menus-section It's just something with the react setup that messes it up for us?

File: src/sidebar/app/components/Footer.js [hunk 1]
```diff
@@ -91,7 +92,6 @@ class Footer extends React.Component {
           });
 
           // Force refresh on Material Design Lite library to activate mdl-menu
```
Comment (vladikoff) [inline src/sidebar/app/components/Footer.js]: delete this line also?

File: src/sidebar/index.html [hunk 2]
```diff
@@ -7,7 +7,6 @@
   <body>
     <div id="notes"></div>
     <script src="vendor/ckeditor.js"></script>
-    <script defer src="vendor/material.js"></script>
```
Comment (vladikoff) [inline src/sidebar/index.html]: remove from `postinstall` and package.json

File: src/sidebar/app/components/Footer.js [hunk 3]
```diff
@@ -286,39 +301,40 @@ class Footer extends React.Component {
...
-            </ul>
+
+          <div className="photon-menu open top left" ref={menu => this.menu = menu }>
+            <button id="context-menu-button" onClick={(e) => this.toggleMenu(e)}>
+              <MoreIcon />
+            </button>
+            <div className="wrapper">
+              <ul role="menu" >
+                <li>
+                  <button
```
Comment (vladikoff) [inline src/sidebar/app/components/Footer.js]: @sebastienbarbier when I was debugging this I noticed that if we remove buttons and just put everything on `<li>` this also works (due to `onClick` being on `<li>` not sure if that simplifies things or not.

Comment (sebastienbarbier) [reply src/sidebar/app/components/Footer.js]: This trigger errors regarding a10y, we need to have click event on clickable object ike button on link.

**Comment by vladikoff** [comment]
@sebastienbarbier https://github.com/google/material-design-lite/blob/mdl-1.x/src/menu/menu.js#L225

**Comment by sebastienbarbier** [comment]
No problem with keyboard, just need to be manually implemented since web default behaviour is using tabulations.

**Comment by sebastienbarbier** [comment]
Also footer looks kinda weird since we have new menu design from latest mockup but old animated login animation. I think we should have footer updated too for 3.1 and then multi-notes will have header and everything else.

**Comment by vladikoff** [comment]
> Also footer looks kinda weird since we have new menu design from latest mockup but old animated login animation. 

Explain? What's weird?

**Comment by vladikoff** [comment]
@sebastienbarbier is it the hover difference ?

**Comment by vladikoff** [comment]
> I think we should have footer updated too for 3.1

Sure, that's fine by me

**Comment by sebastienbarbier** [comment]
darkfooter
Not super happy about my implementation of this dark menu ... but works fine.
@ryanfeeley do we have some specification, or any advice we should apply even if my guess is focusing on light theme 😁

**Comment by sebastienbarbier** [comment]
@vladikoff keyboard navigation is done. Can you review and merge if ok 😁👍

**Comment by Ademd47** [comment]
شكرا لكم

بتاريخ 07‏/03‏/2018 22:56، كتب "Sébastien BARBIER" <notifications@github.com
>:

@vladikoff <https://github.com/vladikoff> keyboard navigation is done. Can
you review and merge if ok 😁👍

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/mozilla/notes/pull/752#issuecomment-371297476>, or mute
the thread

**Comment by vladikoff** [comment]
Let's make the meatball hover as big as sync hover

**Comment by sebastienbarbier** [comment]
largemenubutton
Large button as temporary fix ... footer will be redesigned/refactored with multi note (already visible in `multi-notes` branch)

**Comment by vladikoff** [comment]
@sebastienbarbier this is almost ready:

2 things:

1. Remove the CSS outline dots on click: 



See if you can just do the HOVER effect on keyboard TAB.

2. The dark menu should still be white, compare with Photon menus here:

image

**Comment by sebastienbarbier** [comment]
Should be good 👍
nooutlineonwhitemenu

**Comment by sebastienbarbier** [comment]
🎉

**Pull request merged**